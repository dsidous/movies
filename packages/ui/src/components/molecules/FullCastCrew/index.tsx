import React from 'react';
import { Box, Typography } from '@material-ui/core';

import { Cast } from '@atoms/index';
import { Movie_detailed } from '@typesRoots/movie';
import useStyles from './styles';

interface Props {
  movie: Movie_detailed;
}

const FullCastCrew: React.FC<Props> = ({
  movie: {
    credits: { crew, cast },
  },
}) => {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <div className={classes.col}>
        <Typography variant='h6'>
          <Box fontWeight={700} my={2}>
            Cast
            <span>{` ${cast.length}`}</span>
          </Box>
        </Typography>
        {cast.map((person) => (
          <Cast cast={person} type='full' key={person.cast_id} />
        ))}
      </div>
      <div className={classes.col}>
        <Typography variant='h6'>
          <Box fontWeight={700} my={2}>
            Crew
            <span>{` ${crew.length}`}</span>
          </Box>
        </Typography>
        {crew.map((person) => (
          <Cast cast={person} type='full' key={person.credit_id} />
        ))}
      </div>
    </div>
  );
};

export default FullCastCrew;
