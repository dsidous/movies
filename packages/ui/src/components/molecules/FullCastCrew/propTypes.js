import { object } from 'prop-types';

export const propTypes = {
  movie: object.isRequired,
};

export const defaultProps = {};
