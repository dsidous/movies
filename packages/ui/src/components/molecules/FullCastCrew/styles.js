import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      display: 'grid',
      gridGap: '20px',
      gridTemplateColumns: '1fr',
      margin: '0 auto',
      padding: '0 20px',

      [theme.breakpoints.up('sm')]: {
        gridTemplateColumns: '1fr 1fr',
      },
    },

    col: {
      display: 'grid',
      gridAutoRows: 'min-content',
      gridGap: theme.spacing(1),

      '& h3 span': {
        color: '#999',
        fontSize: '.75em',
      },
    },

    item: {
      cursor: 'pointer',
      display: 'grid',
      gridTemplateColumns: '1fr 2fr',
      gridGap: '20px',
      padding: '10px 0',

      [theme.breakpoints.up('sm')]: {
        gridTemplateColumns: '1fr 4fr',
      },

      '& img': {
        borderRadius: 4,
        width: '100%',
      },
    },

    copy: {
      alignSelf: 'center',
    },
  }),
  { name: 'fullCastCrew' },
);
