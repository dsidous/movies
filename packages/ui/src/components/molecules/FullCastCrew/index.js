import React from 'react';

import { propTypes } from './propTypes';
import Cast from '../../atoms/Cast';
import MediaImage from '../../atoms/MediaImage';

const FullCastCrew = props => {
  const {
    movie: {
      credits: { crew, cast },
    },
  } = props;

  return (
    <div>
      <div className="full-ca-cr">
        <div className="full-ca-cr__col">
          <h3>
            Cast
            <span>{` ${cast.length}`}</span>
          </h3>
          {cast.map(person => (
            <Cast cast={person} type="full" key={person.credit_id} />
          ))}
        </div>
        <div className="full-ca-cr__col">
          <h3>
            Crew
            <span>{` ${crew.length}`}</span>
          </h3>
          {crew.map(person => (
            <div key={person.credit_id} className="full-ca-cr__element">
              <MediaImage
                mediaType="miniProfile"
                filePath={person.profile_path}
                name={person.name}
                className="full-ca-cr__image"
              />
              <div className="full-ca-cr__copy">
                <strong>{person.name}</strong>
                <br />
                <small>{person.job}</small>
              </div>
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};

FullCastCrew.propTypes = propTypes;

export default FullCastCrew;
