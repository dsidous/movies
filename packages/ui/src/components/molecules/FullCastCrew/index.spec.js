import React from 'react';
import { shallow } from 'enzyme';
import FullCastCrew from '.';

const mockProps = {
  movie: {
    credits: {
      crew: [
        {
          credit_id: 1,
          profile_path: 'path/to/image',
          name: 'name',
          job: 'job',
        },
      ],
      cast: [
        {
          credit_id: 1,
        },
      ],
    },
  },
};

describe('Molecules/FullCastCrew', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<FullCastCrew {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
