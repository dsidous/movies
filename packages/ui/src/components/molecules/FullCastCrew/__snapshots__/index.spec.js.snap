// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/FullCastCrew should render as expected 1`] = `ShallowWrapper {}`;
